cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -16,46 +16,146 @@
+import co.cask.cdap.app.runtime.ProgramController;
+import co.cask.cdap.app.runtime.ProgramRuntimeService;
+import co.cask.cdap.app.store.preview.PreviewStore;
 import co.cask.cdap.common.NotFoundException;
+import co.cask.cdap.internal.app.deploy.ProgramTerminator;
+import co.cask.cdap.internal.app.runtime.AbstractListener;
+import co.cask.cdap.internal.app.services.ApplicationLifecycleService;
+import co.cask.cdap.internal.app.services.ProgramLifecycleService;
+import co.cask.cdap.proto.BasicThrowable;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
 import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.proto.artifact.ArtifactSummary;
 import co.cask.cdap.proto.id.ApplicationId;
+import co.cask.cdap.proto.id.ArtifactId;
 import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.ProgramId;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.inject.Inject;
 import org.apache.twill.api.logging.LogEntry;
+import org.apache.twill.common.Threads;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import javax.annotation.Nullable;
 
 /**
  * Default implementation of the {@link PreviewManager}.
  */
 public class DefaultPreviewManager implements PreviewManager {
+  private static final Gson GSON = new Gson();
+  private static final ProgramTerminator NOOP_PROGRAM_TERMINATOR = new ProgramTerminator() {
+    @Override
+    public void stop(ProgramId programId) throws Exception {
+      // no-op
+    }
+  };
+  private final ApplicationLifecycleService applicationLifecycleService;
+  private final ProgramLifecycleService programLifecycleService;
+  private final PreviewStore previewStore;
+  private PreviewStatus status;
+  private AppRequest request;
+
+  @Inject
+  DefaultPreviewManager(ApplicationLifecycleService applicationLifecycleService,
+                        ProgramLifecycleService programLifecycleService, PreviewStore previewStore) {
+    this.applicationLifecycleService = applicationLifecycleService;
+    this.programLifecycleService = programLifecycleService;
+    this.previewStore = previewStore;
+    this.status = null;
+  }
+
   @Override
-  public ApplicationId start(NamespaceId namespaceId, AppRequest<?> request) throws Exception {
-    return null;
+  public void start(ApplicationId preview, AppRequest<?> request) throws Exception {
+    this.request = request;
+    ArtifactSummary artifactSummary = request.getArtifact();
+    NamespaceId artifactNamespace = ArtifactScope.SYSTEM.equals((artifactSummary.getScope())) ? NamespaceId.SYSTEM
+      : preview.getParent();
+
+    ArtifactId artifactId = new ArtifactId(artifactNamespace.getNamespace(), artifactSummary.getName(),
+                                           artifactSummary.getVersion());
+
+    String config = request.getConfig() == null ? null : GSON.toJson(request.getConfig());
```

Comment:
I think deploy method takes config as a json string because the type of config object depends on the Application. In `InMemoryConfigurator` we convert the json into appropriate appConfig object.
https://github.com/caskdata/cdap/blob/develop/cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java#L152